instrumentation/opentelemetry_req/lib/opentelemetry_req.ex

```diff
@@ -30,39 +108,82 @@ defmodule OpentelemetryReq do
   )
   ```
 
-  ### Example without path_params
+  Request and response header attributes are opt-in and can be set with the
+  `request_header_attrs` and `response_header_attrss` options. Values should be lower-case.
+
+  ### Trace Header Propagation
 
+  By default, trace propagation headers are not injected to requests. There are
+  two options available to propagate trace headers:
+
+    * set `propagate_trace_headers` option to `true` when attaching or in the call
+    * manually use `:otel_propagator_text_map.inject/1`
+
+  Example:
   ```
   client =
     Req.new()
-    |> OpentelemetryReq.attach(
-      base_url: "http://localhost:4000",
-      propagate_trace_ctx: true,
-      no_path_params: true
-    )
+    |> OpentelemetryReq.attach(propagate_trace_headers: true)
 
-  client
-  |> Req.get(
-    url: "/api/users"
-  )
+  # or
+
+  client =
+    Req.new()
+    |> OpentelemetryReq.attach()
+
+  Req.get(client, "/", propagate_trace_headers: true)
   ```
-  If you don't set `path_params` the request will raise.
-  """
 
-  alias OpenTelemetry.Tracer
-  alias OpenTelemetry.SemanticConventions.Trace
-  require Trace
-  require Tracer
-  require Logger
+  ### Span Name Override
+
+  The span name can be overridden by setting the `span_name` option in the call.
+
+  Example:
+  ```
+  client =
+    Req.new()
+    |> OpentelemetryReq.attach()
 
+  Req.get(client, "/", span_name: "custom")
+  ```
+
+  > #### Option Precedence {: .info}
+  >
+  > Options passed in a request take precedence over those passed in `attach/1`.
+  >
+  """
+
+  @spec attach(Req.Request.t(), options()) :: Req.Request.t()
   def attach(%Req.Request{} = request, options \\ []) do
+    opts = NimbleOptions.validate!(options, @options_schema) |> Enum.into(%{})
+
+    config =
+      opts
+      |> Enum.into(%{})
+      |> Map.update!(:opt_in_attrs, fn attrs ->
+        attrs
+        |> Enum.filter(&elem(&1, 1))
+        |> Keyword.keys()
+      end)
+      |> then(fn config ->
+        if Enum.member?(opts.opt_in_attrs, {URLAttributes.url_template(), true}) do
+          Map.put(config, :url_template_enabled, true)
+        else
+          Map.put(config, :url_template_enabled, false)
+        end
```

Comment:
What's the deal with renaming this one in particular? ([cross-reference to SemConv](https://github.com/open-telemetry/opentelemetry-erlang/blob/main/apps/opentelemetry_semantic_conventions/lib/incubating/attributes/url_attributes.ex#L265))

Reply:
Not sure I follow. Renaming of what?

Reply:
It looks like SemConv defines this field as `:"url.template"`, but this code is pulling that value out and putting it in the `:url_template_enabled` key in `config`. I think I just misread this as renaming it in the attributes themselves, but it's actually just setting a key in the configuration based on that attribute name, so it's fine! 👍

Reply:
It's just an internal config key to not have to parse the attrs on every request, essentially a precalculation.